[22454] Fix improperly escaped toolbars

oliverguenther opened the pull request:
The user link of `/admin/users/new` was improperly escaped, as was several other toolbars due to the change in https://github.com/opf/openproject/pull/4052. This PR also
- provides a `breadcrumb_toolbar` for the `foo >> bar` style toolbars
- removes the title attribute of a toolbar, since I argue it does not make sense within h2.

oliverguenther committed:
Fix improperly escaped user toolbar

TeatroIO commented:
I've prepared a stage to preview changes. <a href='http://oliverguenther-fix-user-toolbar.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com/' target='_blank'>Open stage</a> or <a href='http://oliverguenther-fix-user-toolbar.opf-openproject-2af22bf12c7dfdad4ddf.ttrcloud.com//teatro.log' target='_blank'>view logs</a>.

oliverguenther committed:
Add link_to option to toolbar title

oliverguenther committed:
Fix all toolbar titles

oliverguenther committed:
Create proper breadcrumb helper to avoid escaping

Lots of toolbars use the notation `<Foobar> >> <Subitem>`,
which did not have a helper yet and requires manually calling html_safe
all over the place.

This fixes it by adding a `breadcrumb_toolbar` helper that properly and
safely joins its components.

oliverguenther commented:
Still working on the plugins toolbars, thus setting to WIP

oliverguenther commented:
Related to https://community.openproject.org/work_packages/22454/activity
